cmake/bintools/arcmwdt/target.cmake

```diff
@@ -0,0 +1,356 @@
+    "RESULT_CMD_LIST;RESULT_BYPROD_LIST;TARGET_INPUT;TARGET_OUTPUT;GAP_FILL;SREC_LEN;FILE_INPUT;FILE_OUTPUT"
+    # List of argument names with multible values
+    "SECTION_ONLY;SECTION_RENAME;SECTION_REMOVE"
+    # Parser input
+    ${ARGN}
+  )
+
+  # Verify arguments
+  if(NOT DEFINED BINTOOLS_OBJCOPY_RESULT_CMD_LIST OR NOT DEFINED ${BINTOOLS_OBJCOPY_RESULT_CMD_LIST})
+    message(FATAL_ERROR "RESULT_CMD_LIST is required.")
+  elseif(NOT DEFINED BINTOOLS_OBJCOPY_FILE_INPUT OR NOT DEFINED BINTOOLS_OBJCOPY_FILE_OUTPUT)
+    message(FATAL_ERROR "Both FILE_INPUT and FILE_OUTPUT is required.")
+  endif()
+
+  # Handle stripping
+  if (DEFINED BINTOOLS_OBJCOPY_STRIP OR DEFINED BINTOOLS_OBJCOPY_STRIP_ALL)
+    if(${BINTOOLS_OBJCOPY_STRIP_ALL})
+      set(obj_copy_strip "-qs")
+    elseif(${BINTOOLS_OBJCOPY_STRIP_DEBUG})
+      set(obj_copy_strip "-ql")
+    endif()
+    set(obj_copy_cmd
+        COMMAND ${CMAKE_STRIP} ${obj_copy_strip}
+        ${BINTOOLS_OBJCOPY_FILE_INPUT} ${BINTOOLS_OBJCOPY_FILE_OUTPUT})
+  endif()
+
+  # no support of --srec-len in mwdt
+
+  # Handle Input and Output target types
+  if(DEFINED BINTOOLS_OBJCOPY_TARGET_OUTPUT)
+    set(obj_copy_target_output "")
+    set(obj_copy_gap_fill "")
+    if(DEFINED BINTOOLS_OBJCOPY_GAP_FILL)
+    set(obj_copy_gap_fill "-f;${BINTOOLS_OBJCOPY_GAP_FILL}")
+    endif()
+      # only mwdt's elf2hex supports gap fill
+    if(${BINTOOLS_OBJCOPY_TARGET_OUTPUT} STREQUAL "srec")
+      set(obj_copy_target_output "-m")
+    elseif(${BINTOOLS_OBJCOPY_TARGET_OUTPUT} STREQUAL "ihex")
+      set(obj_copy_target_output "-I")
+    elseif(${BINTOOLS_OBJCOPY_TARGET_OUTPUT} STREQUAL "binary")
+      set(obj_copy_target_output "-B")
+    endif()
+     set(obj_copy_cmd
+        COMMAND ${CMAKE_ELF2HEX} ${obj_copy_gap_fill} ${obj_copy_target_output}
+        -o ${BINTOOLS_OBJCOPY_FILE_OUTPUT} ${BINTOOLS_OBJCOPY_FILE_INPUT})
+  endif()
+
+  # Handle sections, if any
+  # 1. Section only selection(s)
+  set(obj_copy_sections_only "")
+  if(DEFINED BINTOOLS_OBJCOPY_SECTION_ONLY)
+    foreach(section_only ${BINTOOLS_OBJCOPY_SECTION_ONLY})
+      list(APPEND obj_copy_sections_only "-sn;${section_only}")
+    endforeach()
+    set(obj_copy_cmd
+        COMMAND ${CMAKE_ELF2BIN} -q ${obj_copy_sections_only}
+        ${BINTOOLS_OBJCOPY_FILE_INPUT} ${BINTOOLS_OBJCOPY_FILE_OUTPUT})
+  endif()
+
+ # no support of rename sections in mwdt, here just use arc-elf32-objcopy temporarily
+  set(obj_copy_sections_rename "")
+  if(DEFINED BINTOOLS_OBJCOPY_SECTION_RENAME)
+    foreach(section_rename ${BINTOOLS_OBJCOPY_SECTION_RENAME})
+      if(NOT ${section_rename} MATCHES "^.*=.*$")
+        message(FATAL_ERROR "Malformed section renaming. Must be from=to, have ${section_rename}")
+      else()
+        list(APPEND obj_copy_sections_rename "--rename-section;${section_rename}")
+      endif()
+    endforeach()
+    set(obj_copy_cmd
+        COMMAND arc-elf32-objcopy ${obj_copy_sections_rename}
```

Comment:
> Could host objcopy be used instead?

That should most certainly work on Linux / MacOS, but might be a little trickier on Windows, cause there are lot of Windows developers who doesn't install a native GNU Toolchain/bintools.

>Maybe instead of hardcoding the tool here, add a function to find it/resolve the path and warn if not found?

agree, no hard coding, cause that may lead to failure during build time with a cryptic message, instead of CMake configure phase.
Also a much better error message can be provided to the user.
So something like:
```
find_program(CMAKE_OBJCOPY objcopy)
find_program(CMAKE_ARC_OBJCOPY arc-elf32-objcopy)
if(CMAKE_OBJCOPY)
   ... do stuff
elseif(CMAKE_ARC_OBJCOPY)
   ... do other stuff
else()
  message(FATAL_ERROR "A good error message")
endif()
```